examples/compositor/wl_compositor.c

```diff
@@ -8,7 +8,7 @@
 
 static void destroy_surface_listener(struct wl_listener *listener, void *data) {
 	struct wlr_surface *surface = wl_resource_get_user_data(data);
-	struct wl_compositor_state *state = surface->compositor_data;
```

Comment:
Yes, leave it how it was before, with special fields in wlr_surface for the compositor.